server/src/test/java/org/opensearch/common/settings/SettingTests.java

```diff
@@ -329,12 +329,18 @@ public void testRegexValidator() throws Exception {
         String expectedRegex = "\\d+";
         Pattern expectedPattern = Pattern.compile(expectedRegex);
         RegexValidator regexValidator = new RegexValidator(expectedRegex);
+        RegexValidator regexValidatorMatcher = new RegexValidator(expectedRegex,false);
```

Comment:
This looks great for testing the `false` case, but can we add a test for the true case?

Also, I suspect spotless is going to complain when CI is run here.  Make sure you always run `./gradlew spotlessApply` before pushing your commits here.